propsigner/client.go

```diff
@@ -0,0 +1,66 @@
+package propsigner
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/filecoin-project/go-state-types/crypto"
+	"github.com/filecoin-project/specs-actors/actors/builtin/market"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	pb "github.com/textileio/go-auctions-client/gen/wallet"
+)
+
+const (
+	maxResponseMessageSize = 100 << 10
+)
+
+// RequestSignatureV1 request a signature for a deal proposal to a remote wallet.
+func RequestSignatureV1(
+	ctx context.Context,
+	h host.Host,
+	authToken string,
+	proposal market.DealProposal,
+	remoteWallet peer.ID) (*crypto.Signature, error) {
+	proposalCborBytes := &bytes.Buffer{}
+	if err := proposal.MarshalCBOR(proposalCborBytes); err != nil {
+		return nil, fmt.Errorf("marshaling deal proposal to cbor: %s", err)
+	}
+
+	req := pb.ProposalSigningRequest{
+		AuthToken:            authToken,
+		FilecoinDealProtocol: filecoinDealProtocolV1,
```

Comment:
We indicate in the message which deal protocol this proposal corresponds to. This allows the receiving end to know which `market.DealProposal` struct unmarshal to.

Currently, there's only one `DealProposal` struct in Filecoin, but here we're covering for future newer versions that might have another struct. In that case, we'd send `filecoinDealProtocolV2` (with whatever value it has), and the receiving end would know it should be unmarshaled to that other struct.

In short, mostly future-proof deal proposal structs.